app/models/product_test.rb

```diff
@@ -104,6 +103,23 @@ def archive_records
     save
   end
 
+  def sample_and_duplicate_records(recs, ids, random: Random.new)
+    car = ::Validators::CalculatingAugmentedRecords.new(measures, records, id)
```

Comment:
I believe the records in the initialize for Smoking Gun Validator is only being used for the name comparision. This isn't needed for the validate_calcuated_results methods. Could possibly be changed to

car = ::Validators::CalculatingAugmentedRecords.new(measures, [], id)